snuba/cli/bootstrap.py

```diff
@@ -106,11 +108,28 @@ def bootstrap(
     # tables or distributed tables, etc.
 
     # Create the tables for every dataset.
+    existing_tables = {row[0] for row in conn.execute("show tables")}
     for name in DATASET_NAMES:
         dataset = get_dataset(name)
 
         logger.debug("Creating tables for dataset %s", name)
+        run_migrations = False
         for statement in dataset.get_dataset_schemas().get_create_statements():
-            logger.debug("Executing:\n%s", statement)
-            ClickhousePool().execute(statement)
+            if statement.table_name not in existing_tables:
+                # This is a hack to deal with Materialized views.
+                # All our DDL create statements are in the for of CREATE IF NOT EXISTS
+                # This works well for all tables, except for Materialized views.
+                # It seems Clockhouse parses the SQL query that defines the view even if
+                # the materialized view already exists.
+                # This breaks if we are making a change to the mat view schema.
+                # In that case we cannot run the CREATE statement if the matview already
+                # exists before we ensure all migrations have ran.
```

Comment:
I'm still do not think that I understand what is happening here. Is this actually failing because the materialized view already exists? Or is this failing because the materialized view is referencing a column in the source table that has not yet been migrated? (The latter seems like the case after re-reading the PR description.)

Reply:
it is the latter. Will rephrase again.